p2p/protocol/identify/peer_loop.go

```diff
@@ -0,0 +1,209 @@
+package identify
+
+import (
+	"context"
+	"errors"
+	"sync"
+	"time"
+
+	"github.com/libp2p/go-libp2p-core/helpers"
+	"github.com/libp2p/go-libp2p-core/network"
+	"github.com/libp2p/go-libp2p-core/peer"
+	"github.com/libp2p/go-libp2p-core/protocol"
+
+	pb "github.com/libp2p/go-libp2p/p2p/protocol/identify/pb"
+
+	ggio "github.com/gogo/protobuf/io"
+)
+
+var errProtocolNotSupported = errors.New("protocol not supported")
```

Comment:
is this error not already declared somewhere generic?

Reply:
@Stebalien Do we such errors declared anywhere ? I know of none.

Reply:
https://github.com/multiformats/go-multistream/blob/bd94c78329f9d3cd394f09e53af6138dd5b4db89/client.go#L15

Reply:
Don't want a dependency on multi-stream for this. Copying for now.